skyportal/models.py

```diff
@@ -641,4 +656,109 @@ class FollowupRequest(Base):
+    photometry = relationship('Photometry', back_populates='thumbnails')
+    obj = relationship('Obj', back_populates='thumbnails', uselist=False,
+                       secondary='photometry',
+                       passive_deletes=True)
+
+
+class ObservingRun(Base):
+
+    instrument_id = sa.Column(
+        sa.ForeignKey('instruments.id', ondelete='CASCADE'), nullable=False
+    )
+    instrument = relationship(
+        'Instrument', cascade='save-update, merge, refresh-expire, expunge',
+        uselist=False, back_populates='observing_runs'
+    )
+
+    # name of the PI
+    pi = sa.Column(sa.Text)
+    observers = sa.Column(sa.Text)
+
+    sources = relationship(
+        'Obj', secondary='join(ClassicalAssignment, Obj)',
+        cascade='save-update, merge, refresh-expire, expunge',
+        passive_deletes=True
+    )
+
+    # let this be nullable to accommodate external groups' runs
+    group = relationship('Group', back_populates='observing_runs')
+    group_id = sa.Column(sa.ForeignKey('groups.id', ondelete='CASCADE'),
+                         nullable=True)
+
+    # the person who uploaded the run
+    owner = relationship('User', back_populates='observing_runs')
+    owner_id = sa.Column(sa.ForeignKey('users.id', ondelete='CASCADE'),
+                         nullable=False, index=True)
+
+    assignments = relationship('ClassicalAssignment', passive_deletes=True)
+    calendar_date = sa.Column(sa.Date, nullable=False, index=True)
+
+    @property
+    def _calendar_noon(self):
+        observer = self.instrument.telescope.observer
+        year = self.calendar_date.year
+        month = self.calendar_date.month
+        day = self.calendar_date.day
+        hour = 12.
+        noon = datetime(year=year, month=month, day=day, hour=hour,
+                                 tzinfo=observer.timezone)
+        noon = noon.utctimetuple()
+        return noon
+
+    @property
+    def sunset(self):
+        observer = self.instrument.telescope.observer
+        noon = self._calendar_noon
+        sunset = observer.sun_set_time(noon, which='next')
+        return sunset
+
+    @property
+    def sunrise(self):
+        observer = self.instrument.telescope.observer
+        noon = self._calendar_noon
+        sunrise = observer.sun_rise_time(noon, which='next')
+        return sunrise
+
+
+User.observing_runs = relationship(
+    'ObservingRun', cascade='save-update, merge, refresh-expire, expunge'
+)
+
+
+class ClassicalAssignment(Base):
+
+    requester = relationship('User', back_populates='assignments')
+    requester_id = sa.Column(sa.ForeignKey('users.id', ondelete='CASCADE'),
+                             nullable=False, index=True)
+
+    obj = relationship('Obj', back_populates='assignments')
+    obj_id = sa.Column(sa.ForeignKey('objs.id', ondelete='CASCADE'),
+                       nullable=False, index=True)
+
+    comment = sa.Column(sa.String())
+    status = sa.Column(sa.String(), nullable=False, default="pending")
```

Comment:
should probably constrain the vocabulary around `status` (such as `pending`, `completed`, `in progress`, `removed`, ...).

Reply:
I.e., enum? Should work fine for classical assignments, but each telescope API for robotic assignments will have a unique set of status codes.

Reply:
I have set this to "completed" "not observed" and "pending" in #599